Output escaping

simonwheatley opened the pull request:
See #228

Commit:
WS Cleanup

Commit:
Moar WS cleanup

Commit:
Lil moar WS cleanup

Commit:
Switch from wrappers back to standard hook functions

We're planning to deconstruct the Babble_Plugin class and this change is a small step in that direction.

Commit:
Remove hook wrapper functions from class-plugin.php

simonwheatley committed:
Merge branch 'develop' into remove/simplify-hooks

* develop:
  Add .gitignore file
  Remove end PHP tags to comply with VIP scanner
  WS cleanup

Conflicts:
	class-plugin.php (fixed using "mine")

simonwheatley committed:
Lose the closing PHP tag

Commit:
Fix errantly transposed priority value

Commit:
Add some output escaping

Commit:
Add more output escaping

Commit:
WS cleanup

Commit:
Add some more output escaping

Commit:
Remove a wee bit of overly zealous sanitization

hewsut commented:
This branch was branched from remove/simplify-hooks so it should be much simpler once #214 is merged. I recommend that merge first.

Commit:
Add some missed output escaping

Commit:
Some more WS cleanup

Commit:
Escape all translations echoed directly out

We want to reduce the potential attack vector present from the way WP does localization. Escaping also prevents an errant quote in a translation from breaking HTML syntax.

See: https://vip.wordpress.com/2014/06/20/the-importance-of-escaping-all-the-things/

Commit:
Remove early escaping

WP standard is to escape output as late as possible. We follow that for consistency (predictablity).

Commit:
One too many esc_html adjustments

We don't need to early escape up here. We escape it all later during output.

simonwheatley committed:
Merge branch 'develop' into remove/simplify-hooks

* develop: (27 commits)
  Only run the linter on one version of WordPress, for speed.
  Add PHP Linting
  Ignore PHPStorm
  Updated screenshots
  No need to run `composer install` during Travis testing because we're not using anything from it at the moment. Fixes tests for PHP 5.2.
  Increase the minimum support version of WordPress to 4.0. See #272.
  Move the language file installation to a method in the base test class.
  Update `test_home_url()` so it's now passing.
  Introduce a URL test class. `test_home_url()` is currently failing with a false negative.
  Introduce a Babble-specific base test class.
  Avoid a PHP notice in `Babble_Post_Public::get_transid()`.
  `bbl_switch_to_lang()` and its child methods should check the language code being switched to and return a boolean.
  Avoid a PHP notice in `Babble_Languages::get_url_prefix_from_code()`.
  Correct the buggy test install script which was generated by WP-CLI 0.19.0.
  Test suite readmes.
  Introduce some user role tests.
  Remove the mostly false-negative-generating `test_tested_up_to()`.
  Update phpunit config.
  Update Travis config.
  Readme: Fix typo
  ...

simonwheatley commented:
Addresses #244

simonwheatley committed:
Merge branch 'develop' into remove/simplify-hooks

* develop:
  Correctly exit with non-zero if there are syntax errors when testing on Travis. See #286.
  Ignore Composer lock and vendor directory
  Avoid throwing an exception in `get_new_term_translation_url()`, and return a `WP_Error` instead. See #218
  Multisite phpunit config file.
  Remove the External Update API library and Babble's integration with it. Fixes #242.
  Remove the `wpdb->prefix` prefix from the content and interface language cookies. Implement path-specific cookies instead. Fixes #232.
  Multisite phpunit config file.
  Remove two blocks of unused code.
  Switch to the `rewrite_rules_array` filter. Fixes #233.
  Introduce test coverage for requests. This will cover changes to rewrite rule filtering.
  Make term links respect the term's language instead of always using the current language.
  Allow `Babble_Post_Public::sync_post_meta()` to accept an `$origin_post` argument, so save unnecessary calls to `bbl_get_post_in_lang()`.
  With multiple requests on one page load (eg. during unit tests), we need to make some tweaks to the assumptions made about the current language, URL, and object translations.
  Prep work for request unit tests.
  Use wp_safe_redirect for local redirects
  Use wp_cache_set instead of wp_cache_add
  pt_PT translation files
  Consistency in plugin naming
  When translation jobs are queued, publish an empty post for each language Babble will use the default language content for these posts, thereby showing default language content for other language contexts until the translation is completed.
  The WP SEO Focus Keyword doesn't actually appear on the front end

Conflicts:
	class-languages.php
	class-taxonomy.php
	class-updates.php

simonwheatley committed:
Merge branch 'remove/simplify-hooks' into fix/output-escaping

* remove/simplify-hooks: (47 commits)
  Correctly exit with non-zero if there are syntax errors when testing on Travis. See #286.
  Ignore Composer lock and vendor directory
  Avoid throwing an exception in `get_new_term_translation_url()`, and return a `WP_Error` instead. See #218
  Multisite phpunit config file.
  Remove the External Update API library and Babble's integration with it. Fixes #242.
  Remove the `wpdb->prefix` prefix from the content and interface language cookies. Implement path-specific cookies instead. Fixes #232.
  Multisite phpunit config file.
  Remove two blocks of unused code.
  Switch to the `rewrite_rules_array` filter. Fixes #233.
  Introduce test coverage for requests. This will cover changes to rewrite rule filtering.
  Make term links respect the term's language instead of always using the current language.
  Allow `Babble_Post_Public::sync_post_meta()` to accept an `$origin_post` argument, so save unnecessary calls to `bbl_get_post_in_lang()`.
  With multiple requests on one page load (eg. during unit tests), we need to make some tweaks to the assumptions made about the current language, URL, and object translations.
  Prep work for request unit tests.
  Only run the linter on one version of WordPress, for speed.
  Add PHP Linting
  Ignore PHPStorm
  Updated screenshots
  No need to run `composer install` during Travis testing because we're not using anything from it at the moment. Fixes tests for PHP 5.2.
  Increase the minimum support version of WordPress to 4.0. See #272.
  ...

Conflicts:
	class-locale.php
	external-update-api/external-update-api/euapi.php

simonwheatley commented:
Closing in favour of #296
